tutorials/introductory/getting_started.py

```diff
@@ -0,0 +1,842 @@
+# and layout formatting.
+fig, (ax1, ax2) = plt.subplots(1, 2,
+                               sharey='row',
+                               figsize=[8, 4],
+                               constrained_layout=True)
+
+# Explicit Figure object has separate method for title.
+fig.suptitle('Personal Financial Tracking \'09 - \'10')
+
+# First explicit Axes object plots data with additional keyword arguments.
+ax1.plot(x, y1, label='Income')
+ax1.plot(x, y2, label='Checking')
+ax1.plot(x, y3, color='green', label='Savings')
+
+# First explicit Axes object uses separate methods for ticks on X-Axis,
+# title, and legend. Keyword arguments are for additional configurations.
+ax1.set_xticks(months)
+ax1.set_xticklabels(months, rotation=270)
+ax1.set_title('2009', fontsize='small')
+ax1.legend(loc='upper left')
+
+# Explicit second Axes object plots data similarly to first explicit Axes.
+ax2.plot(x, y1, label='Income')
+ax2.plot(x, y4, label='Checking')
+ax2.plot(x, y5, color='green', label='Savings')
+
+# Explicit second Axes object has separate methods as well.
+ax2.set_xticks(months)
+ax2.set_xticklabels(months, rotation=270)
+ax2.set_title('2010', fontsize='small')
+
+# The pyplot module displays Figure.
+plt.show()
+
+##############################################################################
+#
+# The explicit example above also uses two Axes to graph the data. However,
+# the explicit approach refers to an explicitly generated Axes after creating
+# both the Figure and Axes.
+#
+# In the unpacking process, numerous Axes are assigned to a single variable.
+# To reference a specific Axes, indexing the location of the respective Axes
+# as a matrix through the single variable works as well.
+#
+# The code below demonstrates indexing multiple Axes::
+#
+#   fig, ax = plt.subplots(2,2)
+#
+#   ax[0,0].bar([1,2,3],[1,2,3])
+#   ax[0,1].plot([3,2,1],[3,2,1])
+#   ax[1,0].hist(hist_data)
+#   ax[1,1].imshow([[1,2], [2,1]])
+#
+#
+# The method `matplotlib.figure.Figure.subplot_mosaic` also generates Axes in
+# a layout with contextual names. The link contains more info for using the
+# method.
+#
+# See code example below::
+#
+#   fig = plt.figure()
+#   ax_dict = fig.subplot_mosaic([['bar', 'plot'],
+#                                ['hist', 'image']])
+#
+#   ax_dict['bar'].bar([1,2,3],[1,2,3])
+#   ax_dict['plot'].plot([3,2,1],[3,2,1])
+#   ax_dict['hist'].hist(hist_data)
+#   ax_dict['image'].imshow([[1,2], [2,1]])
+#
+# Implicit
+# ^^^^^^^^
+#
+# There are limitations for customizing the implicit approach without
+# referencing specific Axes and Artists within the Figure. For more advanced
+# configurations, the explicit approach offers more flexibility and control.
+# The Matplotlib community recommends using explicit programming for these
+# tasks.
+#
+# Generalized Function Guidelines
+# -------------------------------
+#
+# For users that have recurring plots and graphs, the signature function
```

Comment:
What is a signature function?

Reply:
I took the language from the [Usage Guide doc](https://matplotlib.org/tutorials/introductory/usage.html#the-object-oriented-interface-and-the-pyplot-interface) for a "signature function" being a template for how to plot for repeated use